internal/controller/promotions/promotions.go

```diff
@@ -117,7 +116,7 @@ func SetupReconcilerWithManager(
 
 	reconciler := newReconciler(
 		kargoMgr.GetClient(),
-		libEvent.NewRecorder(ctx, kargoMgr.GetScheme(), kargoMgr.GetClient(), cfg.Name()),
+		k8sevent.NewKubernetesEventSender(libEvent.NewRecorder(ctx, kargoMgr.GetScheme(), kargoMgr.GetClient(), cfg.Name())),
```

Comment:
I did this for compatibility, but I was wondering if I should actually have the public constructor for this (and other controller things I changed it in) take an `event.Sender` as an argument so it has to be passed in (and can be overridden) at the time of creation